File: c/hamming/src/hamming.c
#include "hamming.h"
#include <string.h>

int hamming_distance(const char *strand_a, const char *strand_b);

int compute(const char *strand_a, const char *strand_b)
{
	if (!strand_a || !strand_b)
		return -1;

	if (strlen(strand_a) != strlen(strand_b))
		return -1;

	return hamming_distance(strand_a, strand_b);
}

int hamming_distance(const char *strand_a, const char *strand_b)
{
	int i, hamming_distance = 0;

	for (i = 0; strand_a[i]; i++)
		if (strand_a[i] != strand_b[i])
			hamming_distance++;

	return hamming_distance;
}

File: python/hamming/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
	"""
	Calculates number of differences (hamming distance) between two equal
	length strings. Raises ValueError if the strings are different lengths
	"""
	if len(strand_a) != len(strand_b):
		raise ValueError("Strands must be same length.")

	return calculate_distance(strand_a, strand_b)


def calculate_distance(strand_a: str, strand_b: str) -> int:
	return sum([a != b for a, b in zip(strand_a, strand_b)])
